shell/components/Resource/Detail/Card/StateCard/composables.ts

```diff
@@ -0,0 +1,111 @@
+import { extractCounts, Props as ResourceRowProps } from '@shell/components/Resource/Detail/ResourceRow.vue';
+import { useI18n } from '@shell/composables/useI18n';
+import { INGRESS, SERVICE } from '@shell/config/types';
+import { getHighestAlertColor } from '@shell/utils/style';
+import { computed, Ref, toValue } from 'vue';
+import { useStore } from 'vuex';
+import { Props as StateCardProps } from '@shell/components/Resource/Detail/Card/StateCard/index.vue';
+
+export function usesResourceCardRow(label: string, resources: any[], to?: any): ResourceRowProps {
+  const colors = resources.map((r: any) => r.stateSimpleColor);
+  const states = resources.map((r: any) => r.stateDisplay.toLowerCase());
+
+  return {
+    label,
+    color:  resources.length ? getHighestAlertColor(colors) : undefined,
+    counts: resources.length ? extractCounts(states) : undefined,
+    to
+  };
+}
+
+export interface Pairs {
+    label: string;
+    to?: any;
+    resources: any[];
+}
+
+export function useDefaultResources(pairs: Ref<Pairs[]>) {
+  const pairsValue = toValue(pairs);
+  const rows = computed(() => pairsValue.map(({ label, resources, to }) => usesResourceCardRow(label, resources, to)));
+
+  return rows;
+}
+
+export function useResourcePair(label: string, type: string, resources?: any[]) {
```

Comment:
Many of these composables can benefit from doc strings to detail usage. It's not immediately clear to me what a `ResourcePair` is, but it looks like we're preparing data to be used with a router.

Reply:
I agree. I'd like to take care of this later though. These are still in flux and based on your other comment we may want to move away from composables.